src/sentry/crons/consumers/checkin.py

```diff
@@ -0,0 +1,61 @@
+import logging
+from typing import Mapping
+
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.commit import CommitOffsets
+from arroyo.processing.strategies.run_task import RunTask
+from arroyo.types import Commit, Message, Partition
+from django.db import transaction
+
+from sentry.models import CheckInStatus, Monitor, MonitorCheckIn, MonitorStatus
+from sentry.utils import json
+from sentry.utils.dates import to_datetime
+
+logger = logging.getLogger(__name__)
+
+
+def process_message(message: Message[KafkaPayload]) -> None:
+    params = json.loads(message.payload.value)
+    current_datetime = to_datetime(params["date_updated"])
+
+    # TODO: Same as MonitorCheckInDetailsEndpoint.put. Keep in sync or factor out.
+    with transaction.atomic():
+        try:
+            checkin = MonitorCheckIn.objects.select_related("monitor").get(
```

Comment:
I think you mentioned in slack, but we'll definitely need this to do upsert.

Tbh you can leave this part to us and we can just get this consumer merged in and we can get everything else in relay going.

Reply:
Thanks, that sounds good to me. We would just need to define the ingest payload for upsert upfront (given that normalization and validation will move to Relay).